pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -44,225 +44,235 @@ type NetworkInfoReconciler struct {
-			return common.ResultRequeueAfter10sec, err
+		if err := r.Service.DeleteVPC(*nsxVPC.Path); err != nil {
+			log.Error(err, "failed to delete VPC in NSX", "VPC", nsxVPC.Path)
+			deleteErrs = append(deleteErrs, fmt.Errorf("failed to delete VPC %s: %w", *nsxVPC.Path, err))
 		}
+	}
+	if len(deleteErrs) > 0 {
+		return fmt.Errorf("multiple errors occurred while deleting VPCs: %v", deleteErrs)
+	}
 
-		var privateIPs []string
-		var vpcConnectivityProfilePath string
-		var nsxLBSPath string
-		isPreCreatedVPC := vpc.IsPreCreatedVPC(nc)
-		if isPreCreatedVPC {
-			privateIPs = createdVpc.PrivateIps
-			vpcConnectivityProfilePath = *createdVpc.VpcConnectivityProfile
-			// Retrieve NSX lbs path if Avi is not used with the pre-created VPC.
-			if createdVpc.LoadBalancerVpcEndpoint == nil || createdVpc.LoadBalancerVpcEndpoint.Enabled == nil ||
-				!*createdVpc.LoadBalancerVpcEndpoint.Enabled {
-				nsxLBSPath, err = r.Service.GetLBSsFromNSXByVPC(*createdVpc.Path)
-				if err != nil {
-					log.Error(err, "failed to get NSX LBS path with pre-created VPC", "VPC", createdVpc.Path)
-					updateFail(r, ctx, obj, &err, r.Client, nil)
-					return common.ResultRequeueAfter10sec, err
-				}
-			}
-		} else {
-			privateIPs = nc.PrivateIPs
-			vpcConnectivityProfilePath = nc.VPCConnectivityProfile
-		}
+	// Update the VPCNetworkConfiguration Status
+	ncName, err := r.Service.GetNetworkconfigNameFromNS(ns)
+	if err != nil {
+		return fmt.Errorf("failed to get VPCNetworkConfiguration for Namespace when deleting stale VPCs %s: %w", ns, err)
+	}
+	deleteVPCNetworkConfigurationStatus(ctx, r.Client, ncName, staleVPCs, r.Service.ListVPC())
+	return nil
+}
 
-		snatIP, path, cidr := "", "", ""
+func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	startTime := time.Now()
+	defer func() {
+		log.Info("Finished reconciling NetworkInfo", "NetworkInfo", req.NamespacedName, "duration", time.Since(startTime))
+	}()
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, common.MetricResTypeNetworkInfo)
 
-		vpcConnectivityProfile, err := r.Service.GetVpcConnectivityProfile(&nc, vpcConnectivityProfilePath)
-		if err != nil {
-			log.Error(err, "get VpcConnectivityProfile failed, would retry exponentially", "VPC", req.NamespacedName)
-			updateFail(r, ctx, obj, &err, r.Client, nil)
-			return common.ResultRequeueAfter10sec, err
-		}
-		hasExternalIPs := true
-		if ncName == commonservice.SystemVPCNetworkConfigurationName {
-			if len(vpcConnectivityProfile.ExternalIpBlocks) == 0 {
-				hasExternalIPs = false
-				log.Error(err, "there is no ExternalIPBlock in VPC ConnectivityProfile", "VPC", req.NamespacedName)
+	networkInfoCR := &v1alpha1.NetworkInfo{}
+	if err := r.Client.Get(ctx, req.NamespacedName, networkInfoCR); err != nil {
+		// IgnoreNotFound returns nil on NotFound errors.
+		if client.IgnoreNotFound(err) == nil {
+			if err := r.deleteStaleVPCs(ctx, networkInfoCR.GetNamespace()); err != nil {
```

Comment:
For this deletion process, we will have the same problem for subnet CR, since when you received the request, 
there is no more networkInfoCR, how can you get namespace from `networkInfoCR.GetNamespace`
you should get ns from `req.NamespacedName`

```
if err := r.deleteSubnetByName(req.Name, req.Namespace); err != nil {
				log.Error(err, "failed to delete NSX Subnet", "Subnet", req.NamespacedName)
```

Do we also need to add combined Namespace tag on VPC object in order to get UID from NSX VPC store. 
Also for distinguishing the two VPC objects when they have same ns/name in the store?

Reply:
Updated.